lib/Listener/BeforeUserLoggedOutListener.php

```diff
@@ -58,12 +59,15 @@ public function handle(Event $event): void {
 
 		$sessionIds = $this->talkSession->getAllActiveSessions();
 		foreach ($sessionIds as $sessionId) {
-			$room = $this->manager->getRoomForSession($user->getUID(), $sessionId);
-			$participant = $room->getParticipant($user->getUID());
-			if ($participant->getInCallFlags() !== Participant::FLAG_DISCONNECTED) {
-				$room->changeInCall($participant, Participant::FLAG_DISCONNECTED);
+			try {
+				$room = $this->manager->getRoomForSession($user->getUID(), $sessionId);
+				$participant = $room->getParticipant($user->getUID());
+				if ($participant->getInCallFlags() !== Participant::FLAG_DISCONNECTED) {
+					$room->changeInCall($participant, Participant::FLAG_DISCONNECTED);
+				}
+				$room->leaveRoom($user->getUID(), $sessionId);
+			} catch (RoomNotFoundException $e) {
```

Comment:
Maybe also catch ParticipantNotFoundException ?

Reply:
It should not be needed, because as `getParticipant` is called immediately after `getRoomForSession` with the same user ID [the participant is "cached"](https://github.com/nextcloud/spreed/blob/21b0d4655936fcc071cbe5ef9733743e8c7d0262/lib/Manager.php#L529) and [the exception would never be thrown](https://github.com/nextcloud/spreed/blob/0de6b875829abbaefc57c1de793fdaac3257fab1/lib/Room.php#L324). But catching it will not hurt either, so :+1: